packages/opentelemetry-exporter-zipkin/test/node/zipkin.test.ts

```diff
@@ -15,7 +15,7 @@
  */
 
 import * as assert from 'assert';
-import * as nock from 'nock';
+import nock from 'nock';
```

Comment:
I chosen a random place for this comment as it applies to a lot places:
Why is `nock` import changed here but not `assert`? At other places `assert` is changed. Is there any systematic or how does someone know what should be used? could we use a lint rule to enforce the "correct" variant?

Reply:
The key point here is that namespace imports can not be used as functions.

When `esModuleInterop` is not enabled, this works but is not spec compliant:
```javascript
import * as assert from 'assert';
// transpiled to
const assert = require('assert'); // `assert` is a function.

assert(true); // works fine
```

In standard ECMAScript module code, or when `esModuleInterop` is enabled:
```javascript
import * as assertNs from 'assert'; // in es spec, this `assertNs` must be an object.
assertNs(true); // assertNs is not callable

import assert from 'assert'; // this is identical to `const assert = require('assert');`
assert(true); // works fine

assertNs !== assert; // `assertNs` is a generated namespace object.
```

So, I just changed the files that using `assert` as functions, but not other files that using it as a namespace for `assert.strictEqual`. Same to `nock`, all `nock` imports are treated as a function, so they have to be changed to default import to get the function value.

Generally, we should use default import like `import assert from 'assert'`. It works with Node.js built-in ECMAScript module support, in `.mjs` files. And it looks much simpler. However, there is no "correct" variant, both two patterns have their own use case. Like, say [here](https://github.com/open-telemetry/opentelemetry-js/blob/main/packages/opentelemetry-sdk-trace-base/src/Span.ts#L17) we do like to import package `@opentelemetry/api` as a namespace to prevent from writing lots of named imports.